src/main/java/com/elixr/poc/rest/controller/PurchasePostController.java

```diff
@@ -0,0 +1,35 @@
+package com.elixr.poc.rest.controller;
+
+import com.elixr.poc.rest.request.PurchaseRequest;
+import com.elixr.poc.rest.response.PurchaseExceptionHandler;
+import com.elixr.poc.service.PurchaseService;
+import jakarta.validation.Valid;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/*controller class for create purchase using post mapping*/
+
+@RestController
+@Validated
+@RequestMapping("/controller")
+public class PurchasePostController {
+
+    private final PurchaseService purchaseService;
+    private PurchaseExceptionHandler exceptionHandler;
+    private PurchaseRequest purchaseRequest;
+
+    public PurchasePostController(PurchaseService purchaseService) {
+        this.purchaseService = purchaseService;
+    }
+
+    @PostMapping("/purchase")
+    public ResponseEntity<?> addPurchase(@RequestBody @Valid PurchaseRequest newPurchase) {
+        purchaseRequest = newPurchase;
```

Comment:
Why newPurchase is assigned to purchaseRequest ?

Reply:
actually, by mistake, i left this line.
REMOVED.